lib/util/batch.js

```diff
@@ -121,7 +121,25 @@ function supportsCql31(version) {
   return version.major > 3 || (version.major === 3 && version.minor >= 1);
 }
 
-var endsWithSemicolonRegex = new RegExp(/\s*;\s*$/);
+function containsDeleteOrUpdateStatement(query) {
+  var match = false;
+  if (query instanceof Batch) {
```

Comment:
Using `instanceof` on custom prototypes makes working with that prototype more difficult. For example I cannot write a Facade which wraps `Batch` because it isn't the exact same prototype. I wouldn't go as far as to say "`instanceof` considered harmful" (although [other folks have](http://perfectionkills.com/instanceof-considered-harmful-or-how-to-write-a-robust-isarray/)), but I generally only use it with prototypes Native to the language. 

I know `instanceof` is used in many places across `priam`, so consistency trumps the technicaly